pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -270,20 +270,33 @@ func (r *Reconciler) delete(ctx context.Context, instance *servingv1alpha1.Knati
 	if len(instance.GetFinalizers()) == 0 || instance.GetFinalizers()[0] != finalizerName {
 		return nil
 	}
+
 	logger.Info("Deleting resources")
-	var RBAC = mf.Any(mf.ByKind("Role"), mf.ByKind("ClusterRole"), mf.ByKind("RoleBinding"), mf.ByKind("ClusterRoleBinding"))
-	if len(r.servings) == 0 {
-		if err := r.config.Filter(mf.ByKind("Deployment")).Delete(); err != nil {
+
+	// List all KnativeEventings to determine if cluster-scoped resources should be deleted.
+	kes, err := r.knativeServingClientSet.OperatorV1alpha1().KnativeServings("").List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all KnativeServings: %w", err)
+	}
+
+	if len(kes.Items) == 1 {
```

Comment:
Same question above applies, but also makes me wonder if we can factor this "delete the RBAC last" logic into a shared lib.

Reply:
We probably can, yes. This is part of a larger refactor which I wanted to get through first before consolidating too much.